client/get_transitions.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+type TransitionsResult struct {
+	Transitions []TransitionObj `json:"transitions"`
+}
+type TransitionObj struct {
+	TransitionId   string `json:"id"`
+	TransitionName string `json:"name"`
+}
+
+func GetTransitions(issueId string) (TransitionsResult, error) {
+	result := TransitionsResult{Transitions: []TransitionObj{}}
+	path := fmt.Sprintf("/rest/api/2/issue/%s/transitions", issueId)
+	request, err := constructGetRequest(path)
+	if err != nil {
+		return result, err
+	}
+	statusCode, err := SendRequest(request, &result)
+	if err != nil {
+		return result, err
+	}
+	if statusCode != http.StatusOK {
+		return result, errors.New("Issue does not exist")
```

Comment:
Is non 200 response code always caused by "Issue does not exist"?  Maybe add a condition check for 404 that throws "Issue does not exist", and then handle non-200 errors.

Reply:
Based on this https://docs.atlassian.com/software/jira/docs/api/REST/7.6.1/#api/2/issue-getTransitions - It returns only 200 or 404 (if the requested issue is not found or the user does not have permission to view it.)